src/MICore/Transports/PipeTransport.cs

```diff
@@ -168,6 +168,8 @@ public override void Close()
                 {
                     // Ignore errors if logout couldn't be written
                 }
+
+                _writer.Close();
```

Comment:
>_writer.Close(); [](http://example.com/codeflow?start=16&length=16)

I think we want to move up:

```
if (_process != null)
            {
                _process.EnableRaisingEvents = false;
                _process.Exited -= OnProcessExit;
}
```

To the start of `Close` so that we don't think the pipe transport is aborting.